tensorflow/core/kernels/dml_random_ops.cc

```diff
@@ -77,6 +77,36 @@ dml::Expression UniformHalf(dml::Graph& scope, dml::Expression input_state,
   return dml::Reinterpret(result, DML_TENSOR_DATA_TYPE_FLOAT16) - 1.0f;
 }
 
+// Compute a + b where a is a signed type and b is unsigned. Requires the result
+// is representable in the range of a's data type. See SignedAdd from
+// random_distributions.h.
+dml::Expression SignedAdd(dml::Expression a, dml::Expression b) {
+  auto b_div_2 = b / 2;
+  return a + dml::Reinterpret(b_div_2, a.GetOutputDesc().dataType) +
+         dml::Reinterpret(b - b_div_2, a.GetOutputDesc().dataType);
+}
+
+// Produces a uniform distribution of integers in the range [lo_value,
+// hi_value). See UniformDistribution<Generator, int32> from
+// random_distributions.h. Requires lo_value < hi_value.
+dml::Expression UniformInt(dml::Graph& graph, dml::Expression input_state,
+                           int32_t lo_value, int32_t hi_value,
```

Comment:
I suppose this naming was copied verbatim from the CPU code, and that we wouldn't want to proper English "low" and "high" here.

Reply:
Yeah, copied from existing code. I suspect people use "lo" and "hi" since it makes the variable lengths the same, which can be aesthetically pleasing, but then min/max also work. Or perhaps it's just to shorten lines of code to avoid ugly clang formatting. Who knows? :)